### PR TITLE
pythonInterpreters: don't use with pkgs

### PR DESCRIPTION
**Pull request by Artturin**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Artturin** [comment]
this used to fail but now works

```diff
diff --git a/pkgs/development/python-modules/xpybutil/default.nix b/pkgs/development/python-modules/xpybutil/default.nix
index cc574c58241..81a0f39abfe 100644
--- a/pkgs/development/python-modules/xpybutil/default.nix
+++ b/pkgs/development/python-modules/xpybutil/default.nix
@@ -17,6 +17,10 @@ buildPythonPackage rec {
 
   propagatedNativeBuildInputs = [ xcffib ];
 
+  preBuild = ''
+    python3 --version
+  '';
+
   # no tests
   doCheck = false;
 
```

`pkgsCross.aarch64-multiplatform.python3Packages.xpybutil`

```
python3.10-xpybutil> /nix/store/7xrm67jylq3xbz2iczlw13llachbahwf-stdenv-linux/setup: line 87: /nix/store/lnz01g03dzvgkxqfanvvq5h05r46bffk-python3-aarch64-unknown-linux-gnu-3.10.7/bin/python3: cannot execute binary file: Exec format error
```

File: pkgs/development/interpreters/python/default.nix [hunk 1]
```diff
@@ -80,7 +78,7 @@ with pkgs;
           optionalExtensions = cond: as: if cond then as else [];
           python2Extension = import ../../../top-level/python2-packages.nix;
           extensions = lib.composeManyExtensions ((optionalExtensions (!self.isPy3k) [python2Extension]) ++ pkgs.pythonPackagesExtensions ++ [ overrides ]);
-          aliases = self: super: lib.optionalAttrs config.allowAliases (import ../../../top-level/python-aliases.nix lib self super);
+          aliases = self: super: lib.optionalAttrs pkgs.config.allowAliases (import ../../../top-level/python-aliases.nix lib self super);
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/python/default.nix:81]: Shouldn't that be just config as an input?

File: pkgs/development/interpreters/python/default.nix [hunk 2]
```diff
@@ -244,45 +242,45 @@ in {
...
     sourceVersion = {
       major = "7";
       minor = "3";
       patch = "7";
     };
     sha256 = "sha256-Ia4zn09QFtbKcwAwXz47VUNzg1yzw5qQQf4w5oEcgMY=";
     pythonVersion = "3.8";
-    db = db.override { dbmSupport = !stdenv.isDarwin; };
-    python = python27;
+    db = pkgs.db.override { dbmSupport = !stdenv.isDarwin; };
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/python/default.nix:269]:
```suggestion
    db = db.override { dbmSupport = !stdenv.isDarwin; };
```
thats berkeley db

**Review by SuperSandro2000 — COMMENTED** [review]
can't comment on the __splicedPackages changes

File: pkgs/development/interpreters/python/default.nix (hunk 1 above)
Comment (Artturin) [reply pkgs/development/interpreters/python/default.nix]: all pkgs removed

**Comment by FRidh** [comment]
Thank you! Nice to see you digging in these parts.

**Pull request merged**